<!DOCTYPE doctype PUBLIC "-//w3c//dtd html 4.0 transitional//en"><html><head>



  <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">





  <meta name="GENERATOR" content="Mozilla/4.76 [en] (Windows NT 5.0; U) [Netscape]">





  <meta name="Author" content="Jim"><title>Software</title></head><body>



<h1>

<font face="Arial,Helvetica"><font color="#006600">Software</font></font></h1>

<hr size="3">
<font face="Arial,Helvetica"><font color="red">Apologies</font>, most of the programs made available below no longer run on current (Windows 7+) platforms. I've checked, and I think the only currently functional programs are anyway probably the most useful today: Ziheng Yang's BETABINO, no. 2 below, Gwenael Piganeau's RECOMBITEST, no. 6 below, and Thomas Lenormand's cline fitting programs, no. 8 below.</font></font>
<hr size="3">
<font face="Arial,Helvetica">LEGACY SITE:</font>
<p><font face="Arial,Helvetica">From time to time, people have asked me to supply 
  various programmes used in our research or teaching.&nbsp; (More recently I've 
  here added particularly useful but not well known packages from other labs, but 
  whose location I might otherwise forget). The downloadable, self-unpacking programmes 
  below come with notes on how to use them.&nbsp; Apart from (5), they are all DOS-based, 
  but should run in MS-DOS or &quot;Command Prompt&quot; windows on anything up 
  to and including Windows XP. &nbsp;If you are not familiar with DOS commands, 
  you should get help, because the programs will often perform oddly if run in a 
  DOS window from within Windows. &nbsp;It is almost always best to run DOS or command 
  prompt programmes in "full screen" mode. &nbsp;
  
<br>
These programs are supplied as self-unpacking ZIP files. &nbsp;They should be 
  downloaded into their own folder, and then unpacked by running the *.EXE self-unpacking 
  file.</font> 
<p><font face="Arial,Helvetica">1) Randomization test for genetic diversity differences 
  between pairs of populations using allozymes and/or mtDNA.</font> <br>
  <font face="Arial,Helvetica"><b><a href="http://www.ucl.ac.uk/%7Eucbhdjm/bin/perm.exe">PERMUTE.EXE</a></b>&nbsp;(32-bit 
  DOS)</font><font face="Arial,Helvetica">, written by Martin Brookes, and modified 
  by J. Mallet. See also: Brookes, M.I., Graneau, Y.A., King, P., Mallet, J.L.B., 
  Rose, O.C. &amp; Thomas, C.D. (1997).&nbsp; The genetic consequences of artificial 
  introductions and natural colonizations in the British butterfly <i>Plebejus 
  argus</i>. Conservation Biology 11: 648-661. </font></p>

<p><font face="Arial,Helvetica">2) Beta-binomial test for hierarchical analyses 
  of count data, for example testing brood survival data in crosses between taxa.</font> 
  <br>
  <font face="Arial,Helvetica"><b><a href="http://www.ucl.ac.uk/%7Eucbhdjm/bin/betabino/">BETABINO.EXE</a></b>&nbsp;(32-bit 
  Windows Console), written by Ziheng Yang (note, will run on 32-bit systems, 
  Windows 95+ only).&nbsp; See also: Jiggins, C.D., Linares, M., Naisbit, R.E., 
  Salazar, C., Yang, Z. &amp; Mallet, J. (2001). Sex-linked hybrid sterility in 
  a butterfly. Evolution 55: 1631-1638. <b><a href="http://www.ucl.ac.uk/taxome/jim/pap/jigginsevol01.pdf">REPRINT</a></b> 
  .</font> </p>

<p><font face="Arial,Helvetica">3) Programs to demonstrate natural selection and 
  genetic drift.</font> <br>
  <font face="Arial,Helvetica"><a href="http://www.ucl.ac.uk/%7Eucbhdjm/bin/natsel/natpack.exe"><b>NATSEL.EXE</b> 
  and <b>DRIFT.EXE</b></a> (16-bit DOS), written by Jim Mallet.</font></p>



<p><font face="Helvetica, Arial, sans-serif">4) Program to calculate linkage disequilibria 
  for loci of arbitrary dominance. <br>
  <b><a href="http://www.ucl.ac.uk/%7Eucbhdjm/bin/DISEQ.EXE">DIS.EXE</a></b></font><font face="Arial,Helvetica">&nbsp;(16-bit 
  DOS)</font><font face="Helvetica, Arial, sans-serif">, written by Jim Mallet. 
  &nbsp;Dominant, recessive, and codominant 
  loci are allowed (e.g. in colour-pattern loci, AFLP loci, RAPDs etc.). Cytonuclear disequilibria can 
  also be estimated (e.g. mtDNA haplotype vs. allozyme loci). For dominant genes, estimates assume Hardy-Weinberg 
  at each locus. Only two alleles per locus are allowed, so this program is mainly 
  useful for hybrid zone studies or studies of natural hybridization. It would be useful for SNPs but not for microsatellites which have many more than two alleles.&nbsp;The program uses a likelihood-based method 
  equivalent to that in Hill WG (1974) Heredity 33: 229-239. See also:&nbsp; Dasmahapatra, K.K., Blum, M.J., Aiello, A., Hackwell, 
  S., Davies, N., Bermingham, E.P. &amp; Mallet, J. (2002). Inferences from a 
  rapidly moving hybrid zone.&nbsp; Evolution 56: 741-753. <b><a href="http://www.ucl.ac.uk/taxome/jim/pap/dasmahapatra.pdf">REPRINT</a></b></font></p>
<p><font face="Helvetica, Arial, sans-serif">5) Program to randomize mimicry associations 
  among hostplants.<br>
  <b><a href="http://www.ucl.ac.uk/%7Eucbhdjm/bin/MIM.EXE">RANDMIM.EXE</a></b></font><font face="Arial,Helvetica">&nbsp;(32-bit 
  WINDOWS). </font><font face="Helvetica, Arial, sans-serif"> &nbsp;Specially written for Keith Willmott, and extremely 
  specialized; not likely to be useful for anyone else! &nbsp;More details on 
  the "Explanations" worksheet of HOSTMIM.XLS supplied in the package.</font> <font face="Arial,Helvetica">See: </font><font face="Helvetica, Arial, sans-serif">Willmott, K.R. 
  &amp; Mallet, J. (2004). &nbsp;Correlations between adult mimicry and larval 
  hostplants in ithomiine butterflies. &nbsp;Proceedings of the Royal Society 
  of London, Series B 271 (Biology Letters S5): S266-S269.&nbsp;<span style="font-family: helvetica,arial,sans-serif;"><a style="font-weight: bold;" href="http://www.ucl.ac.uk/taxome/jim/pap/willmott04.pdf">REPRINT</a></span></font></p>
<p><span style="font-family: helvetica,arial,sans-serif">6) Program to test for 
  recombination in a DNA sequence or haplotype, by testing for evidence for a 
  correlation between linkage disequilibrium and distance along a sequence. <strong><a href="http://www.lifesci.sussex.ac.uk/CSE/test/index.php">RecombiTEST</a></strong> 
  by <a href="http://www.obs-banyuls.fr/UMR7628/equipes/n5/equipe4/people/gwenael.html">Gwenael 
  Piganeau</a> (online test). Tests are separately done with two standardized 
  measures of disequilibrium (<em>R</em><sup>2</sup>, and with <em>D</em>/<em>D<sub>max</sub></em> 
  , or <em>D</em>', and measure the decline in correlation of the disequilibrium 
  measure with distance along a sequence in a sample of aligned haplotypes. The 
  correlation coefficient is bootstrapped 1000 times to test for significance. 
  Programme automatically deletes from the alignment gaps and sites with more 
  than two bases (This is <em>not</em> from our lab, but I couldn't find the website 
  via an internet search, so I have put the link online partly for our own convenience. 
  As used in: Piganeau, G., &amp; Eyre-Walker. A. 2004. A reanalysis of the indirect 
  evidence for recombination in human mitochondrial DNA. Heredity 92:282-288.)</span></p>
<p><span style="font-family: helvetica,arial,sans-serif">7) G-Test <strong><a href="http://www.ucl.ac.uk/%7Eucbhdjm/bin/GTEST.EXE">G.EXE</a></strong>. 
  Simple calculator for G and chi-square (X-square) values and degrees of freedom 
  for n x n homogeneity test tables (16-bit DOS).</span></p>
<p><span style="font-family: helvetica,arial,sans-serif">8) <strong><a href="http://www.cefe.cnrs.fr/fr/recherche/ee/gee/204-cfit">Cline-fitting</a></strong>, 
  C6, C7 etc. for fitting phenotypes or allelic data, including the the possibility 
  of fitting bimodal hybrid zones for the first time that I know about. <a href="http://www.cefe.cnrs.fr/fr/recherche/ee/gee/800-c/193-thomas-lenormand">Thomas 
  LeNormand</a>.</span> </p>
<hr width="100%" size="2"><font face="Arial,Helvetica"><a href="http://www.ucl.ac.uk/taxome/jim/">To

J. Mallet home page</a></font>

<br>
<font face="Arial,Helvetica"><b>Last updated:</b> 30 March 2014</font> <br>



<br>



<br>

<br>

<br>

</body></html>